### PR TITLE
Add documentation for Act URLs and Act directories

### PR DESCRIPTION
**Pull request by HaraldJoerg**
Hi Act-team,
A few people (including me) trying to understand how Act works and maybe improve it have struggled with getting all types of files served correctly.
The documentation is intended to help with that problem.  Of course, a critical review is welcome!

This Pull Request contains no code, just documentation.

### CODE REVIEW

**Comment by eseyman** [comment]
This looks like an excellent start. Is this a WIP or do you want this to be merged?

**Comment by HaraldJoerg** [comment]
If you don't spot any inaccuracies in the description, I'd be glad if it could be merged!  I'd love to take it as a basis for some tests I intend to write.

**Comment by vanHoesel** [comment]
This documentation frequently refers to PSGI version. The branch you want it to be merged into is the Act-Legacy branch. Therefor, it should not be merged with this one, I'm sorry to say so.

File: lib/Act/Manual/Developer/UrlsAndFiles.pod [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+=head3 Where Act is looking for them
+
+I<Disclaimer:> The author doesn't know the directory structure on any
+productive Act server.  The following description is based on the PSGI
+based implementation, so may be inaccurate.
+
+Act has a "home" directory which is passed as a mandatory environment
+variable C<ACTHOME> to the engine.  All paths are calculated from there.
+We are writing C<"$home"> here for this directory.  Every conference
```
Comment (vanHoesel) [inline lib/Act/Manual/Developer/UrlsAndFiles.pod]: 🛑 I do not see any reason to change that naming scheme here, $ACTHOME is a perfectly understood concept within those ever worked on this code

**Comment by vanHoesel** [comment]
Having checked other parts of the manual, I wonder how much clarification this brings. `Act::Manual::Organizer::Layout` gives already plenty information on what directories are used for what purpose.

If any, than it would be a useful addition to show how URL's are resolved into which paths and in which order, using whatever method inside a handler.

File: lib/Act/Manual/Developer/UrlsAndFiles.pod (hunk 1 above)
Comment (HaraldJoerg) [reply lib/Act/Manual/Developer/UrlsAndFiles.pod]: Fair enough - done and committed.

**Comment by vanHoesel** [comment]
👍

**Comment by HaraldJoerg** [comment]
About the extra clarifications: The document helps me to write tests at the HTTP level in my own developer installation.  The organizer layout doesn't inform about the URL space.  Yet I _think_ it is important that the representation in URL-space is kept as it is today, so a reference beyond "just try" should be in order.  Also, the organizer manual doesn't inform about how the stuff "above" the conference tree is organized.  

_How_ URLs are resolved into paths is indeed important to understand.  However, I can't figure this out for the legacy software, so I can't contribute to the master branch on that topic.  I plan, however, to document this for the PSGI dispatcher in my own fork, with the changes I made to clean up some of its rough edges.  I guess it is yet to be seen when and where such changes will be merged :)

**Pull request merged**